app/helpers/tasklist_helper.rb

```diff
@@ -0,0 +1,54 @@
+module TasklistHelper
+  STATUSES = {
+    :not_started => "Not started",
+    :cannot_start_yet =>  "Cannot start yet",
+    :completed => "Completed",
+    :in_progress => "In progress"
+  }
+
+  STYLES = {
+    :not_started => "govuk-tag--grey",
+    :cannot_start_yet => "govuk-tag--grey",
+    :completed => "",
+    :in_progress => "govuk-tag--blue"
+  }
+
+  def get_subsection_status(subsection_name, case_log, questions)
+    if subsection_name == "declaration"
+      return all_questions_completed(case_log) ? :not_started : :cannot_start_yet
+    end
+
+    return :not_started if questions.all? {|question| case_log[question].blank?}
+    return :completed if questions.all? {|question| case_log[question].present?}
```

Comment:
Think this will get more complicated once we have conditional questions in the form but probably best to keep that as a separate PR/Ticket.

Reply:
Yeah, this is a bit to iffy (👀) for my liking, but hopefully not to difficult to change when it comes to it